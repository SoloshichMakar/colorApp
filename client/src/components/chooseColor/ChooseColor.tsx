import React, { useState } from "react";

import { COLORS } from "../../utils/Constants";
import {
  ColorRadioButtonsContainer,
  ColorCheckmark,
  ColorRadioButton,
  InputRadio,
} from "./style";

interface IChooseColorProps {
  setColor: Function;
}

const ChooseColor: React.FC<IChooseColorProps> = ({setColor}) => {
  const [colorRadioButtons, setColorRadioButtons] = useState(
    COLORS.map((color) => ({
      color,
      isChecked: false,
    }))
  );

  const onRadioSelect = (radioIndex: number) => {
    const newColorRadioButtons = colorRadioButtons.map((RadioButton, index) => {
      const isAlreadyChecked = RadioButton.isChecked;
      const shouldBeChecked = !isAlreadyChecked && radioIndex === index;
      return {
        color: RadioButton.color,
        isChecked: shouldBeChecked,
      };
    });

    if (colorRadioButtons[radioIndex].isChecked) {
      setColor("");
    } else {
      setColor(colorRadioButtons[radioIndex].color);
    }
    setColorRadioButtons(newColorRadioButtons);
  };

  return (
    <ColorRadioButtonsContainer>
      {colorRadioButtons.map((colorRadioButton, index) => (
        <ColorRadioButton
          key={colorRadioButton.color}
          className="choose_colors__container"
        >
          <InputRadio
            name="choose_color"
            type="radio"
            value={colorRadioButton.color}
            onClick={() => onRadioSelect(index)}
            checked={colorRadioButton.isChecked}
          />
          <ColorCheckmark
            className="color_checkmark"
            style={{ backgroundColor: colorRadioButton.color }}
          />
        </ColorRadioButton>
      ))}
    </ColorRadioButtonsContainer>
  );
};
export default ChooseColor
