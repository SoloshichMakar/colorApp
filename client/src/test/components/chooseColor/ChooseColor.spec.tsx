import React from 'react';
import renderer from 'react-test-renderer';

import ChooseColor from '../../../components/chooseColor/ChooseColor';
import {BrowserRouter} from "react-router-dom";


const props = {
    setColor: Function
};


it('renders correctly', () => {
    const tree = renderer
        .create(
            <BrowserRouter>
                <ChooseColor
                 setColor={props.setColor}
                />
            </BrowserRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});