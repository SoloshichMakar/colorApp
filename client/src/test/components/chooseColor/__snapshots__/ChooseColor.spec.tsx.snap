// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm iirJRb"
>
  <label
    className="sc-AxirZ bgLfMu choose_colors__container"
  >
    <input
      checked={false}
      className="sc-AxhCb jQeLEK"
      name="choose_color"
      onClick={[Function]}
      type="radio"
      value="#f16d69"
    />
    <span
      className="sc-AxiKw grTOxR color_checkmark"
      style={
        Object {
          "backgroundColor": "#f16d69",
        }
      }
    />
  </label>
  <label
    className="sc-AxirZ bgLfMu choose_colors__container"
  >
    <input
      checked={false}
      className="sc-AxhCb jQeLEK"
      name="choose_color"
      onClick={[Function]}
      type="radio"
      value="#f279a2"
    />
    <span
      className="sc-AxiKw grTOxR color_checkmark"
      style={
        Object {
          "backgroundColor": "#f279a2",
        }
      }
    />
  </label>
  <label
    className="sc-AxirZ bgLfMu choose_colors__container"
  >
    <input
      checked={false}
      className="sc-AxhCb jQeLEK"
      name="choose_color"
      onClick={[Function]}
      type="radio"
      value="#9170cb"
    />
    <span
      className="sc-AxiKw grTOxR color_checkmark"
      style={
        Object {
          "backgroundColor": "#9170cb",
        }
      }
    />
  </label>
  <label
    className="sc-AxirZ bgLfMu choose_colors__container"
  >
    <input
      checked={false}
      className="sc-AxhCb jQeLEK"
      name="choose_color"
      onClick={[Function]}
      type="radio"
      value="#5eb3f6"
    />
    <span
      className="sc-AxiKw grTOxR color_checkmark"
      style={
        Object {
          "backgroundColor": "#5eb3f6",
        }
      }
    />
  </label>
  <label
    className="sc-AxirZ bgLfMu choose_colors__container"
  >
    <input
      checked={false}
      className="sc-AxhCb jQeLEK"
      name="choose_color"
      onClick={[Function]}
      type="radio"
      value="#67d7e5"
    />
    <span
      className="sc-AxiKw grTOxR color_checkmark"
      style={
        Object {
          "backgroundColor": "#67d7e5",
        }
      }
    />
  </label>
  <label
    className="sc-AxirZ bgLfMu choose_colors__container"
  >
    <input
      checked={false}
      className="sc-AxhCb jQeLEK"
      name="choose_color"
      onClick={[Function]}
      type="radio"
      value="#ffe083"
    />
    <span
      className="sc-AxiKw grTOxR color_checkmark"
      style={
        Object {
          "backgroundColor": "#ffe083",
        }
      }
    />
  </label>
</div>
`;
